ovs/lib/generic.py

```diff
@@ -391,6 +433,14 @@ def _verify_mds_config(current_vdisk):
                                 GenericController._logger.warning('Scrubber - vPool {0} - StorageRouter {1} - vDisk {2} - Skipping because master MDS still not local'.format(vpool.name, storagerouter.name, vdisk.name))
                                 continue
 
+                        # Check if vDisk is already being scrubbed
+                        with volatile_mutex('execute_scrub_work', wait=5):
+                            volatile_key = 'scrub_vdisk_{0}'.format(vdisk_guid)
+                            if volatile_client.get(volatile_key) is not None:
+                                GenericController._logger.warning('Scrubber - vPool {0} - StorageRouter {1} - vDisk {2} - Skipping because vDisk is already being scrubbed'.format(vpool.name, storagerouter.name, vdisk.name))
+                                continue
+                            volatile_client.set(key=volatile_key, value=volatile_key, time=24 * 60 * 60)
```

Comment:
You might be able to get rid of the lock when using `volatile_client.add(...)` instead. This call returns a `False`-ish value when the key already exists. If it doesn't, it adds the key. This is atomic.